ITR/data/base_providers.py

```diff
@@ -569,38 +633,66 @@ def _standardize(self, intensities: pd.DataFrame) -> pd.DataFrame:
-            # See https://github.com/hgrecco/pint-pandas/issues/114
-            winsorized: pd.DataFrame = historic_intensities.clip(
-                # Must set numeric_only to false to process Quantities
-                lower=historic_intensities.quantile(q=self.projection_controls.LOWER_PERCENTILE, axis='index',
-                                                    numeric_only=False),
-                upper=historic_intensities.quantile(q=self.projection_controls.UPPER_PERCENTILE, axis='index',
-                                                    numeric_only=False),
-                axis='columns'
-            )
-        return winsorized
+            # quantile doesn't handle pd.NA inside Quantity; FIXME: we can use np.nan because not expecting UFloat in input data
+            try:
+                # Turns out we have to dequantify here: https://github.com/pandas-dev/pandas/issues/45968
+                # Can try again when ExtensionArrays are supported by `quantile`, `clip`, and friends
+                units = historic_intensities.apply(lambda x: x[x.first_valid_index()].u
+                                                   if x.first_valid_index() and isinstance(x[x.first_valid_index()], pint.Quantity) else None)
+                if ITR.HAS_UNCERTAINTIES:
+                    nominal_intensities = historic_intensities.apply(lambda x: ITR.nominal_values(x.map(lambda y: y.m if isinstance(y, pint.Quantity) else np.nan)))
+                    uncertain_intensities = historic_intensities.apply(lambda x: ITR.std_devs(x.map(lambda y: y.m if isinstance(y, pint.Quantity) else 0)))
+                else:
+                    nominal_intensities = historic_intensities.apply(lambda x: x.map(lambda y: y.m if isinstance(y, pint.Quantity) else np.nan))
+                # See https://github.com/hgrecco/pint-pandas/issues/114
+                winsorized: pd.DataFrame = nominal_intensities.clip(
+                    lower=nominal_intensities.quantile(q=self.projection_controls.LOWER_PERCENTILE, axis='index',
+                                                        numeric_only=False),
+                    upper=nominal_intensities.quantile(q=self.projection_controls.UPPER_PERCENTILE, axis='index',
+                                                        numeric_only=False),
+                    axis='columns'
+                )
+                if ITR.HAS_UNCERTAINTIES:
+                    if uncertain_intensities.values.sum() != 0:
+                        uwinsorized = winsorized.apply(lambda x: PA_(ITR.uarray(x.values.data, uncertain_intensities[x.name].values), dtype=units[x.name]))
+                        return uwinsorized
+                winsorized_and_unitized = winsorized.apply(lambda x: PA_(x.values.data, dtype=units[x.name]))
+                return winsorized_and_unitized
+            except AttributeError:
+                breakpoint()
```

Comment:
should we raise an exception as well?
it's an `AttributeError` - what does it mean, to mute this exception?

Reply:
I've simplified this code to assume everything is OK.  If there's an error, it will raise itself.  But it shouldn't because we've constructed the inputs to the function before calling it.